<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1110</width>
    <height>830</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>110</y>
      <width>931</width>
      <height>681</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Part1</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_openImg1">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>90</y>
        <width>161</width>
        <height>71</height>
       </rect>
      </property>
      <property name="text">
       <string>OpemImg</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_apply">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>530</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Apply</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>0</y>
        <width>320</width>
        <height>320</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Original image</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label2">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>0</y>
        <width>320</width>
        <height>320</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Trapezoidal</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label3">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>330</y>
        <width>320</width>
        <height>320</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Wavy</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label4">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>330</y>
        <width>320</width>
        <height>320</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Circular</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QTabWidget" name="tabWidget_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>221</width>
        <height>291</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Trapezoidal</string>
       </attribute>
       <widget class="QSlider" name="horizontalSlider_x">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>50</y>
          <width>160</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_x_offset">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>30</y>
          <width>58</width>
          <height>15</height>
         </rect>
        </property>
        <property name="text">
         <string>x_offset :</string>
        </property>
       </widget>
       <widget class="QSlider" name="horizontalSlider_y">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>110</y>
          <width>160</width>
          <height>22</height>
         </rect>
        </property>
        <property name="value">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_y_offset">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>90</y>
          <width>58</width>
          <height>15</height>
         </rect>
        </property>
        <property name="text">
         <string>y_offset :</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_x_offset_display">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>30</y>
          <width>21</width>
          <height>21</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(8, 8, 8);</string>
        </property>
        <property name="text">
         <string>50</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_y_offset_display">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>90</y>
          <width>21</width>
          <height>21</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(8, 8, 8);</string>
        </property>
        <property name="text">
         <string>50</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Wavy</string>
       </attribute>
       <widget class="QLabel" name="label_per_display">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>90</y>
          <width>21</width>
          <height>21</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(8, 8, 8);</string>
        </property>
        <property name="text">
         <string>100</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QSlider" name="horizontalSlider_mag">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>50</y>
          <width>160</width>
          <height>22</height>
         </rect>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>25</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QLabel" name="label_mag_display">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>30</y>
          <width>21</width>
          <height>21</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(8, 8, 8);</string>
        </property>
        <property name="text">
         <string>25</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_period">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>90</y>
          <width>41</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>period : </string>
        </property>
       </widget>
       <widget class="QLabel" name="label_mag">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>30</y>
          <width>41</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>mag : </string>
        </property>
       </widget>
       <widget class="QSlider" name="horizontalSlider_per">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>110</y>
          <width>160</width>
          <height>22</height>
         </rect>
        </property>
        <property name="maximum">
         <number>200</number>
        </property>
        <property name="value">
         <number>100</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Part2</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>Part3</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_openImg_3">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>70</y>
        <width>161</width>
        <height>71</height>
       </rect>
      </property>
      <property name="text">
       <string>OpemImg</string>
      </property>
     </widget>
     <widget class="QLabel" name="label6">
      <property name="geometry">
       <rect>
        <x>580</x>
        <y>0</y>
        <width>320</width>
        <height>320</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Hough</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_apply_3">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>460</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>Apply</string>
      </property>
     </widget>
     <widget class="QLabel" name="label5">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>0</y>
        <width>320</width>
        <height>320</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Original image</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_mag_2">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>240</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Binaary threshold</string>
      </property>
     </widget>
     <widget class="QSlider" name="horizontalSlider_threshold">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>260</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>128</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_threshold_display">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>240</y>
        <width>21</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>128</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_size">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>200</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>11</number>
      </property>
      <property name="singleStep">
       <number>2</number>
      </property>
      <property name="value">
       <number>3</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_mag_3">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>150</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Gaussian kernal</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_mag_4">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>300</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Line number</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_line_display">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>290</y>
        <width>21</width>
        <height>21</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="horizontalSlider_line">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>330</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label7">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>330</y>
        <width>320</width>
        <height>320</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Edge image</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label8">
      <property name="geometry">
       <rect>
        <x>580</x>
        <y>330</y>
        <width>320</width>
        <height>320</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(215, 215, 215); border-color: rgb(8, 8, 8);</string>
      </property>
      <property name="text">
       <string>Binary image</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_sigma">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>200</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>11</number>
      </property>
      <property name="singleStep">
       <number>2</number>
      </property>
      <property name="value">
       <number>3</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_mag_5">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>180</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Size</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_mag_6">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>180</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Sigma</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>141</width>
      <height>111</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Adobe Heiti Std</family>
      <pointsize>26</pointsize>
     </font>
    </property>
    <property name="text">
     <string>HW6</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>80</y>
      <width>161</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Palace Script MT</family>
      <pointsize>26</pointsize>
      <italic>true</italic>
     </font>
    </property>
    <property name="text">
     <string>TA so handsome</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1110</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
